### DISCUSSION TITLE
Make Query Panel Resizable

### DISCUSSION THREAD

**Discussion by mertbakir** [grafana/grafana#38333] [Data Sources]
**What would you like to be added**:

I would like to use a resizable query panel.

**Why is this needed**:

Because it's hard to work on long queries in this small panel. It would be even better if we can split screen in half vertically and write query on one side and see the result on the other side. Resizable panel should be easier to implement. You may consider the other option though.

![resizablegrafanaquerypanel](https://user-images.githubusercontent.com/43188411/105228762-00a3c980-5b74-11eb-8488-339ba587cd6b.gif)

**Comment by aocenas** [comment]
Is this specific data source? From the gif I assume SQL datasources in general but want to make sure. I know ideally this would be solved globally but as data sources can be developed differently by different teams it may not be easy to automatically solve in all places

**Comment by mertbakir** [comment]
> Is this specific data source? From the gif I assume SQL datasources in general but want to make sure. I know ideally this would be solved globally but as data sources can be developed differently by different teams it may not be easy to automatically solve in all places

Well, I'm using postgresql but it should be data source independent.

**Comment by aocenas** [comment]
@papagian maybe something that can be quick fixed in the postgres data source.

@jessover9000 As far as I know we don't have any UX guidelines for plugin/datasource development where something like this could be put, do we?

**Comment by jessover9000** [comment]
@aocenas There's no general agreement that I know of as of now. I also checked storybook and for the TextArea component we already have the drag handle. Do we use a separate component for code editors? I think there's no harm in having the same drag handle for any type of multi-row input.

As we're just picking up the work on query editors, we are planning to come up with some overarching guidelines for the future, and I'm pretty sure all multi-row code editors can benefit from such a resizing option.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Reply by ecgg** [reply]
This would be useful for InfluxDB data sources

**Comment by gabor** [comment]
this has been fixed for sql datasources, if you use a new version of Grafana, there is an "expand editor" button in the lower right corner that will maximize the editor.
closing the discussion, if you feel like this solution does not solve your problem, please create a new github issue, and please mentioned there why this solution is not suitable. thanks!
